src/backend/base/langflow/components/data/APIRequest.py

```diff
@@ -111,10 +111,17 @@ async def make_request(
         if method not in ["GET", "POST", "PATCH", "PUT", "DELETE"]:
             raise ValueError(f"Unsupported method: {method}")
 
+        if isinstance(body, str):
+            try:
+                body = json.loads(body)
+            except Exception as e:
+                logger.error(f"Error decoding JSON data: {e}")
```

Comment:
This should raise an exception because if the payload is None then the request will fail. 

Also, the check could be `isinstance(body, str) and body`

Reply:
@ogabrielluiz, thanks for your review!
Just one point, I think I can have a request with the payload as None. Like GETs, PATCHs or DELETEs, the payload is not necessarily required. But I added the and condition on the if as you suggested!